AnkiDroid/src/main/java/com/ichi2/utils/StringUtil.java

```diff
@@ -0,0 +1,40 @@
+/*
+ Copyright (c) 2020 David Allison <davidallisongithub@gmail.com>
+
+ This program is free software; you can redistribute it and/or modify it under
+ the terms of the GNU General Public License as published by the Free Software
+ Foundation; either version 3 of the License, or (at your option) any later
+ version.
+
+ This program is distributed in the hope that it will be useful, but WITHOUT ANY
+ WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License along with
+ this program.  If not, see <http://www.gnu.org/licenses/>.
+
+ */
+
+package com.ichi2.utils;
+
+import org.jetbrains.annotations.Contract;
+
+import androidx.annotation.Nullable;
+
+public class StringUtil {
+
+    /** Trims from the right hand side of a string */
+    @Nullable
+    @Contract("null -> null; !null -> !null")
+    public static String trimRight(@Nullable String s) {
+        if (s == null) {
+            return null;
+        }
+
+        int newLength = s.length();
+        while (newLength > 0 && Character.isWhitespace(s.charAt(newLength - 1))) {
```

Comment:
```suggestion
        while (newLength >= 0 && Character.isWhitespace(s.charAt(newLength - 1))) {
```

'>=' ? Just thinking, an rTrim of a fully whitespace string should return the empty string, so it should be valid to go all the way to zero? It appears substring only throws if endIndex is less than beginIndex so equal index values should work and empty string should be valid

https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#substring(int,%20int)

Reply:
No, if the newLength is 0, then `.charAt(newLength - 1)` will crash, and the decrement operation will take the length to -1.

But, I'll add tests to prove it.

Reply:
Done and re-pushed, No code changes in `StringUtil`

Reply:
oh of course - I was mentally trying to do the index (zero-based) vs length shifts but reasoned through it incorrectly. If newLength goes through at 1 and it's whitespace it will go to 0 in the loop, break the loop then substring 0 0 - I can see it